src/azure_storage_account_client.cpp

```diff
@@ -142,7 +142,12 @@ CreateChainedTokenCredential(const std::string &chain,
 		} else if (item == "workload_identity") {
 			sources.push_back(std::make_shared<Azure::Identity::WorkloadIdentityCredential>(credential_options));
 		} else if (item == "managed_identity") {
-			sources.push_back(std::make_shared<Azure::Identity::ManagedIdentityCredential>(credential_options));
+			if (env_client_id.empty()) {
```

Comment:
TODO: check whether `account_name` is set, as it appears to be required by AZ SDK and leads to failure + stack trace otherwise.